src/EventStore.Projections.Core/Services/Processing/ResultEventEmitter.cs

```diff
@@ -29,6 +41,9 @@ public class ResultEventEmitter : IResultEventEmitter {
 							: new EmittedDataEvent(
 								streamId, Guid.NewGuid(), "Result", true, projectionResult, null, at, null),
 						_resultStreamMetadata);
+				if (!string.IsNullOrEmpty(result.Event.Data) && result.Event.EventType.Equals("Result")) {
```

Comment:
i think this is the same check as below isn't it

Reply:
Yeah it is indeed the same check. The purpose of keeping this check outside of the method CheckStateSize() was to avoid calling the CheckStateSize() method for every emitted event, but rather only call it for the events that determine the size of the projection state (that is events of type "Result").